src/sql/workbench/services/notebook/browser/models/clientSession.ts

```diff
@@ -116,8 +117,8 @@ export class ClientSession implements IClientSession {
 			session.defaultKernelLoaded = true;
 		} catch (err) {
 			// TODO move registration
-			if (err && err.response && err.response.status === 501) {
-				this.options.notificationService.warn(localize('kernelRequiresConnection', "Kernel {0} was not found. The default kernel will be used instead.", kernelSpec.name));
+			if (err?.response?.status === 501 || err?.message === this._jupyterNotImplementedError) {
```

Comment:
I wouldn't be opposed to modifying the error serialization a bit to allow certain properties too. But if this is just a one-off thing it's probably fine I guess...just have to hope that the message never changes or isn't ever going to be localized I guess.

If you plan on keeping this at least add a comment explaining why the second check is necessary.

Reply:
Switched to using a custom errorCode property.